src/config/src/yaml.rs

```diff
@@ -113,6 +113,21 @@ pub trait Element: Sized {
         None
     }
 
+    /// From some string (should be used for top level elements if you want to embed the code)
+    fn from_string<'a>(src: &'a str) -> Result<Self, ConfigError> {
+        let mut meta = ConfigMeta::default();
+        meta.path = PathBuf::from("");
+        let yaml = try!(YamlLoader::load_from_str(src).map_err(|e| ConfigError::YamlScan(e)));
```

Comment:
I think you should stick to the more idiomatic `?` notation instead of `try!`, if possible.